recipes/sail/all/conanfile.py

```diff
@@ -18,27 +18,53 @@ class SAILConan(ConanFile):
         "shared": [True, False],
         "fPIC": [True, False],
         "thread_safe": [True, False],
-        "with_avif": [True, False],
-        "with_gif": [True, False],
-        "with_jpeg2000": [True, False],
-        "with_jpeg": ["libjpeg", "libjpeg-turbo", False],
-        "with_png": [True, False],
-        "with_tiff": [True, False],
-        "with_webp": [True, False],
+        "with_highest_priority_codecs": [True, False],
+        "with_high_priority_codecs": [True, False],
+        "with_medium_priority_codecs": [True, False],
+        "with_low_priority_codecs": [True, False],
+        "with_lowest_priority_codecs": [True, False],
+        "with_avif": [True, False, "deprecated"],
+        "with_gif": [True, False, "deprecated"],
+        "with_jpeg2000": [True, False, "deprecated"],
+        "with_jpeg": ["libjpeg", "libjpeg-turbo", False, "deprecated"],
+        "with_png": [True, False, "deprecated"],
+        "with_tiff": [True, False, "deprecated"],
+        "with_webp": [True, False, "deprecated"],
+
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "thread_safe": True,
-        "with_avif": True,
-        "with_gif": True,
-        "with_jpeg2000": True,
-        "with_jpeg": "libjpeg",
-        "with_png": True,
-        "with_tiff": True,
-        "with_webp": True,
+        "with_highest_priority_codecs": True,
+        "with_high_priority_codecs": True,
+        "with_medium_priority_codecs": True,
+        "with_low_priority_codecs": True,
+        "with_lowest_priority_codecs": True,
+        "with_avif": "deprecated",
+        "with_gif": "deprecated",
+        "with_jpeg2000": "deprecated",
+        "with_jpeg": "deprecated",
+        "with_png": "deprecated",
+        "with_tiff": "deprecated",
+        "with_webp": "deprecated",
+    }
+    options_description = {
+        "with_avif": "Deprecated",
+        "with_gif": "Deprecated",
+        "with_jpeg2000": "Deprecated",
+        "with_jpeg": "Deprecated",
+        "with_png": "Deprecated",
+        "with_tiff": "Deprecated",
+        "with_webp": "Deprecated",
+        "with_highest_priority_codecs": "Enable highest priority codecs",
+        "with_high_priority_codecs": "Enable high priority codecs",
+        "with_medium_priority_codecs": "Enable medium priority codecs",
+        "with_low_priority_codecs": "Enable low priority codecs",
+        "with_lowest_priority_codecs": "Enable lowest priority codecs",
```

Comment:
```suggestion
        "with_highest_priority_codecs": "Enable highest priority codecs",
        "with_high_priority_codecs": "Enable high priority codecs",
        "with_medium_priority_codecs": "Enable medium priority codecs",
        "with_low_priority_codecs": "Enable low priority codecs",
        "with_lowest_priority_codecs": "Enable lowest priority codecs",
```

@HappySeaFox Do see a better way to explain it? Like, which codec could be inserted in each option, or maybe, what refers a specific priority? I searched in https://github.com/HappySeaFox/sail/blob/master/BUILDING.md#cmake-options-overview, but TBH, I didn't find any clue.

Reply:
Hi! Yes, every codec has explicit priority defined. See https://github.com/HappySeaFox/sail/blob/3bddd3f50a710cbf000162de555ef20e4e9a2d0a/src/sail-codecs/CMakeLists.txt#L7, https://github.com/HappySeaFox/sail/blob/3bddd3f50a710cbf000162de555ef20e4e9a2d0a/src/sail-codecs/avif/avif.codec.info.in#L6 as an example

Reply:
Thank you! I just updated the list, according the other codecs.